.github/workflows/rock-update.yaml

```diff
@@ -0,0 +1,110 @@
+name: Update ROCK on new releases of its source
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  workflow_call:
+    inputs:
+      rock-name:
+        description: "Name of the application for which to build the ROCK"
+        required: true
+        type: string
+      source-repo:
+        description: "Repository of the source application in 'org/repo' form"
+        required: true
+        type: string
+      check-go: 
+        description: "Flag to check updates on the Go version"
+        default: false
+        required: false
+        type: boolean
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  check-version:
+    name: Detect new releases
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install dependencies
+        run: |
+          sudo snap install jq
+          sudo snap install yq
+
+      - id: latest-release
+        name: Fetch version used in *latest* release
+        run: |
+          TAG=$(curl -L \
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}"\
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            https://api.github.com/repos/${{ inputs.source-repo }}/releases/latest \
+            | jq -r .tag_name)
+          echo "release=$TAG" >> $GITHUB_OUTPUT
+
+      - name: Checkout the ROCK source
+        uses: actions/checkout@v3
+        with:
+          path: main
+
+      - id: check
+        name: Check if the release has already been added
+        shell: bash
+        run: |
+          source_tag="${{ steps.check.outputs.release }}"
+          version=${source_tag#"v"}
+          if [ ! -f /$version/rockcraft.yaml ]; then
+            echo "version=$version" >> $GITHUB_OUTPUT
+            echo "release=${{steps.latest-release.outputs.release}}" >> $GITHUB_OUTPUT
+            echo "New upstream release ${{steps.latest-release.outputs.release}} found"
+          else
+            echo "No new upstream release found"
+          fi
+
+      - id: check-go-version
+        name: Checkout application source for the Go version check
+        if: ${{ inputs.check-go }} && ${{ steps.check.outputs.release != '' }}
+        uses: actions/checkout@v3
+        with:
+          repository: ${{ inputs.source-repo }}
+          ref: ${{ steps.check.outputs.release }}
+          path: application-src
+      
+      - name: Create a new rockcraft.yaml for the new application version
+        if: ${{ steps.check.outputs.release != '' }}
+        shell: bash
+        run: |
+          source_tag="${{ steps.check.outputs.release }}"
+          version="${{ steps.check.outputs.version }}"
+          mkdir $GITHUB_WORKSPACE/main/$version
+          latest_rockcraft_file=$(find $GITHUB_WORKSPACE/main/ -name "rockcraft.yaml" | sort -V | tail -n1)
```

Comment:
We should probably document at the top of this file + PR description that the action relies on folders being named in a way that `sort -V` understands.

Reply:
I agree about semver tagging won't change, but the folder hierarchy/structure is totally up to us.
Perhaps the doc line could be as simple as:

> This action automatically creates a folder with the same name as the upstream tag